modules/affinity/src/main/java/org/apache/ignite/internal/affinity/Assignments.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.affinity;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Set;
+import org.apache.ignite.internal.tostring.IgniteToStringInclude;
+import org.apache.ignite.internal.tostring.S;
+import org.apache.ignite.internal.util.ByteUtils;
+
+/**
+ * Class that encapsulates a set of nodes and its metadata.
+ */
+public class Assignments implements Serializable {
+    /** Serial version UID. */
+    private static final long serialVersionUID = -59553172012153869L;
+
+    /** Empty assignments. */
+    public static final Assignments NO_ASSIGNMENTS = new Assignments(Collections.emptySet());
+
+    /** Set of nodes. */
+    @IgniteToStringInclude
+    private final Set<Assignment> nodes;
+
+    /**
+     * Constructor.
+     *
+     * @param nodes Set of nodes.
+     */
+    public Assignments(Set<Assignment> nodes) {
+        this.nodes = nodes;
+    }
+
+    /**
+     * Returns a set of nodes, represented by this assignments instance.
+     */
+    public Set<Assignment> nodes() {
+        return nodes;
+    }
+
+    /**
+     * Serializes the instance into an array of bytes.
+     */
+    public byte[] toBytes() {
+        return ByteUtils.toBytes(this);
+    }
+
+    /**
+     * Serializes assignments into an array of bytes.
+     *
+     * @see #toBytes()
+     */
+    public static byte[] toBytes(Set<Assignment> assignments) {
+        return new Assignments(assignments).toBytes();
+    }
+
+    /**
+     * Deserializes assignments from the array of bytes.
+     */
+    public static Assignments fromBytes(byte[] bytes) {
```

Comment:
What do we get in case of 1) empty 2) null bytes?

Reply:
Exceptions in both cases. I can handle nulls I guess

Reply:
I believe that we should return empty assignments in both cases. There's similar code below `entry.value() == null? NO_ASSIGNMENS : Assignments.fromBytes(entry.value())`. I believe that we should hide it insider fromBytes.